osu.Game/Skinning/Components/BeatmapAttributeText.cs

```diff
@@ -101,6 +130,19 @@ private void updateBeatmapContent(WorkingBeatmap workingBeatmap)
             valueDictionary[BeatmapAttribute.StarRating] = workingBeatmap.BeatmapInfo.StarRating.ToLocalisableString(@"F2");
         }
 
+        private void getMaxPP()
+        {
+            // this is a null score to pass mod.
+            ScoreInfo score = new ScoreInfo(beatmap.Value.BeatmapInfo, beatmap.Value.BeatmapInfo.Ruleset) { Mods = currentMods.Value.ToArray() };
```

Comment:
if the score is only manufactured to pass mods, then `CalculatePerfectPerformance` should just accept mods and not a score